CI-API.Data/Repository/AdminRepository.cs

```diff
@@ -330,5 +236,529 @@ private JsonResult updateAddUser(User userHasTobeUpdated,UserDetailViewModel use
+        {
+            try
+            {
+                if (search != null)
+                {
+
+                    List<Mission> AllMission = await Task.FromResult(cIDbContext.Missions.Where(M => M.Title.Contains(search) || M.Theme.Title.Contains(search)).ToList());
+                    List<MissionTheme> AllMissionThemes = await Task.FromResult(cIDbContext.MissionThemes.ToList());
+                    List<MissionSkill> AllMissionSkills = await Task.FromResult(cIDbContext.MissionSkills.ToList());
+
+                    LandingPageViewModel landingPageViewModel = new()
+                    {
+                        missions = AllMission,
+                        missionThemes = AllMissionThemes,
+                        missionSkills = AllMissionSkills,
+
+                    };
+
+                    return new JsonResult(new apiResponse<LandingPageViewModel> { StatusCode = responseStatusCode.Success, Data = landingPageViewModel, Result = true });
+                }
+                else
+                {
+                    List<Mission> AllMission = await Task.FromResult(cIDbContext.Missions.ToList());
+                    List<MissionTheme> AllMissionThemes = await Task.FromResult(cIDbContext.MissionThemes.ToList());
+                    List<MissionSkill> AllMissionSkills = await Task.FromResult(cIDbContext.MissionSkills.ToList());
+
+                    LandingPageViewModel landingPageViewModel = new()
+                    {
+                        missions = AllMission,
+                        missionThemes = AllMissionThemes,
+                        missionSkills = AllMissionSkills,
+
+                    };
+
+                    return new JsonResult(new apiResponse<LandingPageViewModel> { StatusCode = responseStatusCode.Success, Data = landingPageViewModel, Result = true });
+                }
+            }
+            catch
+            {
+                return new JsonResult(new apiResponse<string> { Message = ResponseMessages.InternalServerError, StatusCode = responseStatusCode.BadRequest, Result = false });
+
+            }
+        }
+        #endregion
+
+        #region GetMissionDataFromId
+        public async Task<JsonResult> getMissionDataFromId(long? missionId)
+        {
+            try
+            {
+                Mission? missionData = await Task.FromResult(cIDbContext.Missions.Where(M => M.MissionId == missionId).FirstOrDefault());
+                List<MissionDocument> documentsForMission = await Task.FromResult(cIDbContext.MissionDocuments.Where(MD => MD.MissionId == missionId).ToList());
+                List<MissionMedium> mediaForMission = await Task.FromResult(cIDbContext.MissionMedia.Where(MM => MM.MissionId == missionId).ToList());
+                List<MissionSkill> skillForMission = await Task.FromResult(cIDbContext.MissionSkills.Where(MS => MS.MissionId == missionId).ToList());
+                if (missionData?.MissionType == StaticCode.GoalMission)
```

Comment:
Use join query to get related data for mission instead of sending separate DB requests.